python/shotgun_fields/date_and_time_widget.py

```diff
@@ -68,3 +53,82 @@ def _create_human_readable_timestamp(self, dt, postfix=""):
 
         time_str = dt.strftime(format)
         return time_str
+
+    def _string_value(self, value):
+        """
+        Convert the Shotgun value for this field into a string
+
+        :param value: The value to convert into a string
+        :type value: :class:`datetime.datetime` or a float representing unix time
+        """
+
+        # shotgun_model converts datetimes to floats representing unix time so
+        # handle that as a valid value as well
+        if not isinstance(value, datetime.datetime):
+            value = datetime.datetime.fromtimestamp(value)
+        return self._create_human_readable_timestamp(value, " %I:%M%p")
+
+
+class DateAndTimeEditorWidget(QtGui.QDateTimeEdit):
+    """
+    Allows editing of a ``date_time`` field value as returned by the Shotgun API.
+
+    Pressing ``Enter`` or ``Return`` when the widget has focus will cause the
+    value to be applied and the ``value_changed`` signal to be emitted.
+    """
+    __metaclass__ = ShotgunFieldMeta
+    _EDITOR_TYPE = "date_time"
+
+    def get_value(self):
+        """
+        :return: The internal value being displayed by the widget.
+        """
+        value = self.dateTime()
+
+        if hasattr(QtCore, "QVariant"):
+            # pyqt
+            return value.toPyDateTime()
+        else:
+            # pyside
+            return value.toPython()
+
+    def keyPressEvent(self, event):
+        """
+        Provides shortcuts for applying modified values.
+
+        :param event: The key press event object
+        :type event: :class:`~PySide.QtGui.QKeyEvent`
+        """
+
+        if event.key() in [QtCore.Qt.Key_Enter, QtCore.Qt.Key_Return]:
+            self.value_changed.emit()
+        else:
+            super(DateAndTimeEditorWidget, self).keyPressEvent(event)
+
+    def setup_widget(self):
+        """
+        Prepare the widget for display.
+
+        Called by the metaclass during initialization.
+        """
+        self.setCalendarPopup(True)
+        self.setMinimumWidth(100)
```

Comment:
I'd prefer the 100 be assigned to a class variable and then that being referenced here.